### PR TITLE
Task/t317 review namespaces endpoints

### LINKED ISSUE #317: Review namespaces endpoints
In accordance with nemtech/symbol-openapi#107 and using the new pagination

### PR DESCRIPTION
**Pull request by martinayora**
Fixes #317.

### CODE REVIEW

**Comment by dgarcia360** [comment]
@martinayora If it's available on MongoDB should be ok to also add it. Please, let me know if you finally add it to update the OpenAPI accordingly.

**Comment by Vektrat** [comment]
> @martinayora If it's available on MongoDB should be ok to also add it. Please, let me know if you finally add it to update the OpenAPI accordingly.

we will keep the same functionality as before, having it filter implicitly unless a requirement arises in the future

File: catapult-sdk/src/plugins/namespace.js [hunk 1]
```diff
@@ -65,11 +65,19 @@ const namespacePlugin = {
 			namespaces: { type: ModelType.array, schemaName: 'namespaceDescriptor' }
 		});
 
+		builder.addSchema('namespaceDescriptorWithId', {
```
Comment (Vektrat) [inline catapult-sdk/src/plugins/namespace.js:68]: i'm not sure i understand this...

why do we need both `namespaceDescriptor` and `namespaceDescriptorWithId`? both endpoints using those schemas should return `id`, `meta`, and `namespace`, right? it is possible that `/namespaces/:namespaceId` is not parsing the id right now

related: `/namespaces/:namespaceId` is using `findNamesByIds` which uses `sanitizer.deleteIds`, not sure this is correct

Comment (martinayora) [reply catapult-sdk/src/plugins/namespace.js]: You were right, id in the `/namespaces/:namespaceId` endpoint wasn't being parsed. Now this is fixed. Regarding `sanitizer.deleteIds` it was changed to `sanitizer.renameId` in a previous commit.

**Pull request merged**